### PR TITLE
FastStringBuffer improvement

### PR DESCRIPTION
**Pull request by dizzzz**
Improved performance by using native array copy
Synced code from saxonHE
Added tests

### CODE REVIEW

File: exist-core/src/main/java/org/exist/util/FastStringBuffer.java [hunk 1]
```diff
@@ -471,12 +472,12 @@ public String getNormalizedString( int mode ) {
      *@param  sb    Description of the Parameter
      *@param  mode  Description of the Parameter
      *@return       The normalizedString value
-     */    
+     */
     public StringBuffer getNormalizedString(StringBuffer sb, int mode ) {
```
Comment (adamretter) [inline exist-core/src/main/java/org/exist/util/FastStringBuffer.java]: Should this be changed to use `StringBuilder` rather than String buffer?

Comment (dizzzz) [reply exist-core/src/main/java/org/exist/util/FastStringBuffer.java]: the classname is Fast...StringBuffer :-) but it makes sense in a way, I will check the users of the method.

**Comment by dizzzz** [comment]
@adamretter reworked PR, added some additional improvements that might yield in less overhead (less code execution, stack usage, unneeded conversation to objects etc)

**Comment by adamretter** [comment]
Excellent stuff!

**Pull request merged**